Neos.Flow/Classes/Core/Booting/Scripts.php

```diff
@@ -736,9 +741,73 @@ public static function buildPhpCommand(array $settings)
             $command .= ' -c ' . escapeshellarg($useIniFile);
         }
 
+        self::ensureWebSubrequestsUseCurrentlyRunningPhpVersion($command);
+
         return $command;
     }
 
+    /**
+     * Compares the realpath of the configured PHP binary (if any) with the one flow was called with in a CLI request.
+     * This avoids config errors where users forget to set Neos.Flow.core.phpBinaryPathAndFilename in CLI.
+     *
+     * @param string phpBinaryPathAndFilename
+     * @throws Exception
+     */
+    protected static function ensureCLISubrequestsUseCurrentlyRunningPhpBinary($phpBinaryPathAndFilename)
+    {
+        // Do nothing for non-CLI requests
```

Comment:
Why not? Wouldn't it make sense to check this *especially* for commands triggered by a Web request?

Reply:
See below. The check can not be done realiably from Web request, because PHP_BINARY is not reliable in that context. The best one can do is check if the version of the specified php binary matches the current php version.